backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt

```diff
@@ -63,6 +63,13 @@ private operator fun String.times(count: Int): String {
     return builder.toString()
 }
 
+private val KotlinType.shortName
```

Comment:
This is somewhat wrong. Consider the behaviour for types with arguments or nullability.
`predefinedTypes` should probably be `predefinedClasses` (or even `predefinedValueClasses`).

Reply:
Not sure if I agree here. This is pretty much list of types we want default factories for. They do not have to be classes, just happen to be so for primitive types.

Reply:
`shortName` and `createNullableName` properties are still wrong: don't work for arbitrary types. There are different options available here, for example to rename these (e.g. `shortNameForPredefinedType` and `createNullableNameForPredefinedType`).

Reply:
PTAL